v3/src/data-interactive/resource-parser-utils.ts

```diff
@@ -41,3 +44,13 @@ export function parseSearchQuery(query: string, dataContextOrCollection?: IDataS
   
   return { valid, left, right, func }
 }
+
+export function findTileFromV2Id(v2Id: string) {
+  const { document } = appState
+  // We look for every possible v3 id the component might have (because each tile type has a different prefix).
+  // Is there a better way to do this?
```

Comment:
I don't know if there's a better way. Probably worth asking Kirk to weigh in when he returns.

Reply:
This function was introduced in a previous PR, so Kirk has seen it and gave it a 👍 previously. It's not the most elegant function, but since the number of tile types is so low at least it's not terribly inefficient.